packages/slice-machine/src/features/navigation/NavigationItem.tsx

```diff
@@ -0,0 +1,80 @@
+import { ActionListItem, Tooltip, useMediaQuery } from "@prismicio/editor-ui";
+import Link from "next/link";
+import {
+  type FC,
+  type MouseEventHandler,
+  type ReactNode,
+  type SVGProps,
+  useMemo,
+} from "react";
+
+type NavigationItemPropsBase = {
+  title: string;
+  active?: boolean;
+  Icon: FC<SVGProps<SVGSVGElement>>;
+  RightElement?: ReactNode;
+};
+
+type NavigationLinkItemProps = NavigationItemPropsBase & {
+  href: string;
+  target?: "_blank";
+  onClick?: never;
+};
+
+type NavigationButtonItemProps = NavigationItemPropsBase & {
+  href?: never;
+  target?: never;
+  onClick: MouseEventHandler<Element>;
+};
+
+type NavigationItemProps = NavigationLinkItemProps | NavigationButtonItemProps;
+
+export function NavigationItem(props: NavigationItemProps) {
+  const { title, href, target, active, Icon, RightElement, onClick } = props;
+
+  const isCollapsed = useMediaQuery({ max: "medium" });
+
+  const ItemIcon = useMemo(() => {
+    return () => <Icon width={isCollapsed ? 28 : 32} />;
+  }, [Icon, isCollapsed]);
```

Comment:
We really need to suggest fixing this on editor-ui 😅 

Nit: Usually functions/components defined at render time use a `useCallback`.

```suggestion
  const ItemIcon = useCallback(() => {
    return () => <Icon width={isCollapsed ? 28 : 32} />;
  }, [Icon, isCollapsed]);
```

Reply:
changed, thanks!